### PR TITLE
Loki: add a 'since' query parameter for easier querying using relative time

### PR DESCRIPTION
**Pull request by slim-bean**
Signed-off-by: Edward Welch <edward.welch@grafana.com>

**What this PR does / why we need it**:

logcli has a command line parameter called `since` which makes it easy to query logs relative to the current time, however this is some syntactic sugar used in logcli to calculate `start` and `end` query parameters for the query sent to Loki.

This PR adds native support for a `since` query parameter to make it a little easier to consume the Loki query API without having to calculate precise start/end times with every request.

This is useful if you are polling the Loki API from a third party app which doesn't let you easily build dynamic URL's and all you want to do is poll for say the last 3 hours of logs.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the `CONTRIBUTING.md` guide
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/logql/downstream_test.go [hunk 1]
```diff
@@ -34,21 +34,22 @@ func TestMappingEquivalence(t *testing.T) {
...
+		//{`sum by (a) (rate({a=~".+"}[1s]))`, false},
+		//{`sum(rate({a=~".+"}[1s]))`, false},
+		//{`max without (a) (rate({a=~".+"}[1s]))`, false},
+		//{`count(rate({a=~".+"}[1s]))`, false},
+		//{`avg(rate({a=~".+"}[1s]))`, true},
+		//{`avg(rate({a=~".+"}[1s])) by (a)`, true},
+		//{`1 + sum by (cluster) (rate({a=~".+"}[1s]))`, false},
+		//{`sum(max(rate({a=~".+"}[1s])))`, false},
+		//{`max(count(rate({a=~".+"}[1s])))`, false},
+		//{`max(sum by (cluster) (rate({a=~".+"}[1s]))) / count(rate({a=~".+"}[1s]))`, false},
```
Comment (MichelHollands) [inline pkg/logql/downstream_test.go:51]: Why are these commented out?

Comment (slim-bean) [reply pkg/logql/downstream_test.go]: ugh, had some stuff on my working directory I didn't notice when I made this... fixed..

File: pkg/loghttp/params.go [hunk 2]
```diff
@@ -48,14 +48,40 @@ func shards(r *http.Request) []string {
...
+
+func determineBounds(now time.Time, startString, endString, sinceString string) (time.Time, time.Time, error) {
+	since := defaultSince
+	if sinceString != "" {
+		d, err := model.ParseDuration(sinceString)
+		if err != nil {
+			return time.Time{}, time.Time{}, errors.Wrap(err, "could not parse 'since' parameter")
+		}
+		since = time.Duration(d)
+	}
```
Comment (DylanGuedes) [inline pkg/loghttp/params.go:65]: double checking: `parseTimestamp(sinceString, defaultSince)` won't work here?

**Review by DylanGuedes — APPROVED** [review]
lgtm, just a nit/question

**Review by kavirajk — APPROVED** [review]
Nice. LGTM

File: pkg/loghttp/params.go (hunk 2 above)
Comment (owen-d) [reply pkg/loghttp/params.go:65]: Since is a _duration_, not a _time_ so parsing it as a time won't work :(

**Review by owen-d — APPROVED** [review]
LGTM

**Pull request merged**